utils/accessibility.go

```diff
@@ -0,0 +1,172 @@
+package utils
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/chromedp/cdproto/runtime"
+	"github.com/chromedp/chromedp"
+)
+
+const axeVersion = "4.10.0"
+
+type Violation struct {
+	ID          string          `json:"id"`
+	Impact      string          `json:"impact"`
+	Tags        []string        `json:"tags"`
+	Description string          `json:"description"`
+	Help        string          `json:"help"`
+	HelpURL     string          `json:"helpUrl"`
+	Nodes       []ViolationNode `json:"nodes"`
+}
+
+type ViolationNode struct {
+	Impact string `json:"impact"`
+	HTML   string `json:"html"`
+}
+
+type AccessibilityConfig struct {
+	Rules   map[string]Rule `json:"rules,omitempty"`
+	RunOnly RunOnly         `json:"runOnly,omitempty"`
+}
+
+type RunOnly struct {
+	Type   string   `json:"type,omitempty"`
+	Values []string `json:"values,omitempty"`
+}
+
+type Rule struct {
+	Enabled bool `json:"enabled"`
+}
+
+// Exports the Accessibility config as a JSON string to supply to axe-core
+func (cfg *AccessibilityConfig) JSON() (string, error) {
+	cfgJSON, err := json.Marshal(cfg)
+	if err != nil {
+		return "", err
+	}
+	return string(cfgJSON), nil
+}
+
+func provisionAccessibilityTooling(ctx context.Context) error {
+	if !accessibilityToolsExist(ctx) {
+		return addAccessibilityTools(ctx)
+	}
+	return nil
+}
+
+func addAccessibilityTools(ctx context.Context) error {
+	var buf []byte
+
+	provisionScript := fmt.Sprintf(`
```

Comment:
Would this work using the js template? https://pkg.go.dev/html/template#JS

Reply:
oh cool, didn't know this was a thing!

Reply:
had a look at this, not sure what benefit it brings - happy to discuss if you've got some time